pytorch_lightning/core/lightning.py

```diff
@@ -246,7 +246,26 @@ def truncated_bptt_steps(self, truncated_bptt_steps: int) -> None:
     @property
     def logger(self) -> Optional[Logger]:
         """Reference to the logger object in the Trainer."""
-        return self.trainer.logger if self.trainer else None
+        # this should match the implementation of `trainer.logger`
+        # we don't reuse it so we can properly set the deprecation stacklevel
+        if self.trainer is None:
+            return
+        loggers = self.trainer.loggers
+        if len(loggers) == 0:
+            return None
+        if len(loggers) == 1:
+            return loggers[0]
+        else:
+            if not self._running_torchscript:
+                rank_zero_deprecation(
+                    "Using `lightning_module.logger` when multiple loggers are configured."
+                    " This behavior will change in v1.8 when `LoggerCollection` is removed, and"
+                    " `lightning_module.logger` will return the first logger available.",
+                    stacklevel=5,
```

Comment:
Fix 3: we set the proper stacklevel

Reply:
but do we need to set it for this deprecation message?

Reply:
Because it's inside a property, so we want the warning to point out the code calling the property, not the code inside the property.